Training5/thongnq-training5/src/state/reducers/todos.js

```diff
@@ -0,0 +1,41 @@
+import {
+    CREATE_TODO,
+    RETRIEVE_TODOS,
+    UPDATE_TODO,
+    DELETE_TODO,
+    FILTER_TODO_BY_NAME
+} from "../actions/type";
+
+const initialState = [];
+
+const todoReducer = (todos = initialState, action) => {
+    const {type, payload} = action;
+
+    switch (type){
+        case CREATE_TODO:
+            return [...todos, payload];
+
+        case RETRIEVE_TODOS:
+            return payload;
+
+        case UPDATE_TODO:
+            return todos.map(todo => {
+                if(todo.id === payload.id){
+                    return {...todo, ...payload};
+                } else {
+                    return todo;
+                }
+            });
+
+        case DELETE_TODO:
+            return todos.filter(todo => todo.id !== payload.id);
+
+        case FILTER_TODO_BY_NAME:
```

Comment:
@Thong-Nguyen-Zigvy we don't filter todos in the store, store always store all todo fetched, if you filter in here how we revert full list todo when search value clear